torch/csrc/profiler/kineto_shim.cpp

```diff
@@ -316,6 +317,15 @@ void logInvariantViolation(
 namespace autograd {
 namespace profiler {
 c10::DeviceType deviceTypeFromActivity(libkineto::ActivityType activity_type) {
+  // gpu_device is for returning currently used device type for Activity
+  // such as concurrent kernel or gpu memory operations
+  // It should be one of registered non-CPU device, so we can init it with
+  // CPU as a invalid value for checking.
+  c10::DeviceType gpu_device = c10::DeviceType::CPU;
+  if (at::hasCUDA() || at::hasMTIA())
+    gpu_device = c10::DeviceType::CUDA;
+  else if (at::hasXPU())
```

Comment:
I'm not sure how AMD users understand it, but at least XPU users would be extremely confused when they find out that the DeviceType is CUDA instead of XPU.